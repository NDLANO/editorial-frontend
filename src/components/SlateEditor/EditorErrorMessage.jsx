import React from 'react';
import PropTypes from 'prop-types';
import BEMHelper from 'react-bem-helper';

const classes = new BEMHelper({
  name: 'editor',
  prefix: 'c-',
});

const EditorErrorMessage = ({ msg, attributes, children }) =>
  <div>
    <div {...classes('error')} {...attributes}>
      <span>
        {msg}
      </span>
      {children}
    </div>
  </div>;

EditorErrorMessage.propTypes = {
  attributes: PropTypes.shape({
    'data-key': PropTypes.string.isRequired,
  }),
  msg: PropTypes.string.isRequired,
};

export default EditorErrorMessage;
