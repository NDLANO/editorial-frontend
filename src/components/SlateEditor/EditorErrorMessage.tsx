/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { HTMLAttributes, MouseEventHandler, ReactNode } from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { colors } from '@ndla/core';
import { EditorDeleteButton } from './common/EditorDeleteButton';

const StyledEditorErrorMessage = styled.div`
  position: relative;
  border: 1px solid ${colors.support.red};
  color: ${colors.support.red};
  padding: 1rem;
`;

interface Props {
  msg: string;
  onRemoveClick?: MouseEventHandler;
  attributes?: HTMLAttributes<HTMLDivElement>;
  children?: ReactNode;
}

const EditorErrorMessage = ({ msg, attributes, onRemoveClick, children }: Props) => {
  const { t } = useTranslation();

  return (
    <StyledEditorErrorMessage {...attributes} contentEditable={false}>
      {onRemoveClick && (
        <EditorDeleteButton aria-label={t('form.remove')} onClick={onRemoveClick} />
      )}
      <span>{msg}</span>
      {children}
    </StyledEditorErrorMessage>
  );
};

export default EditorErrorMessage;
