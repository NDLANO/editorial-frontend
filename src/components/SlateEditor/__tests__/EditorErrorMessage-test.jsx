/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import TestRenderer from 'react-test-renderer';
import EditorErrorMessage from '../EditorErrorMessage';

test('EditorErrorMessage renders', () => {
  const embed = {
    message: 'This is error',
    resource: 'error',
  };

  const component = TestRenderer.create(
    <EditorErrorMessage embed={embed} msg={embed.message} onRemoveClick={() => {}} />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
