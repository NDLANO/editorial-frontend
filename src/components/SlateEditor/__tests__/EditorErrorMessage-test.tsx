/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { render } from '@testing-library/react';
import EditorErrorMessage from '../EditorErrorMessage';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';

test('EditorErrorMessage renders', () => {
  const embed = {
    message: 'This is error',
    resource: 'error',
  };

  const { container } = render(
    <IntlWrapper>
      <EditorErrorMessage msg={embed.message} onRemoveClick={() => {}} />
    </IntlWrapper>,
  );

  expect(container).toMatchSnapshot();
});
