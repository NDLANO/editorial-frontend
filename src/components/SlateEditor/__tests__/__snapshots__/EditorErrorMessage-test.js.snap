// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorErrorMessage renders 1`] = `
<div
  className="css-121c9wz emotion-1"
>
  <button
    className="css-15w35ia emotion-0"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="c-icon"
      data-license="Apache License 2.0"
      data-source="Material Design"
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <title>
        Cross
      </title>
      <g>
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </g>
    </svg>
  </button>
  <span>
    This is error
  </span>
</div>
`;
